global_logger.py

```diff
@@ -241,8 +243,15 @@ def logged_class_method(wrapped, level: Optional[int] = None) -> classmethod:
     return _LoggedWrapperType.CLASS_METHOD.build_wrapper(wrapped, level)
 
 
-def log_error(e: BaseException, msg=''):
-    logger.error(f'{msg} - {e!r}')
+def log_error(e: Exception, msg=''):
+    import traceback
+    tb = traceback.format_tb(e.__traceback__)
+    tb_lines = []
+    for item in tb:
+        tb_lines.extend(item.split('\n'))
+    logger.error(f'{msg} - {e!r}' if msg else repr(e))
+    for line in tb_lines:
+        logger.error(f'    {line}')
```

Comment:
`log_error` is way better now. It will print the traceback which has been super helpful to me when trying to trouble shoot issues.